unittests/unit/test_ConstraintSolver.cpp

```diff
@@ -81,3 +89,24 @@ TEST(ConstraintSolver, SIMPLE)
   skel->computeImpulseForwardDynamics();
   EXPECT_TRUE(box->getRelativeSpatialVelocity()[4] >= 0);
 }
+
+TEST(ConstraintSolver, ReplaceConstraintEngineWithFrictionlessLcp)
+{
+  auto world = loadWorldAndIntegrateNonConstraintForces();
+  auto skel = world->getSkeleton("box skeleton");
+  auto box = skel->getBodyNode("box");
+  auto solver = world->getConstraintSolver();
+
+  world->replaceConstraintEngineFn([world](bool _resetCommand) {
+      return world->runFrictionlessLcpConstraintEngine(_resetCommand);
+    });
+  world->runRegisteredConstraintEngine(false);
+
+  // Collision detection.
+  EXPECT_TRUE(solver->getLastCollisionResult().getNumContacts() > 0);
+  EXPECT_TRUE(solver->getNumConstrainedGroups() > 0);
+
+  // Integrate velocities from solved impulses. Should have non-negative normal
+  // velocity after integration.
+  EXPECT_TRUE(box->getRelativeSpatialVelocity()[4] >= 0);
```

Comment:
I think we need one more check here to check that having no friction actually made a difference in this case. Maybe we apply a small horizontal velocity before we run the constraints engine, and check that it's still there afterwards?

It would make sense to apply the same small horizontal velocity to the test for the LCP with friction as well, and check that the friction ends up zeroing out the velocity.